layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteNavigationRestService.java

```diff
@@ -192,6 +200,48 @@ public Response moveNode (@Parameter(description = "node id") @QueryParam("nodeI
     }
   }
 
+  @Path("{pageRef}")
+  @PUT
+  @RolesAllowed("users")
+  @Operation(summary = "update page access abd edit permission", method = "PUT", description = "update page access abd edit permission")
+  @ApiResponses(value = { @ApiResponse(responseCode = "204", description = "Request fulfilled"),
+          @ApiResponse(responseCode = "400", description = "Invalid query input"),
+          @ApiResponse(responseCode = "404", description = "Page not found"),
+          @ApiResponse(responseCode = "401", description = "Unauthorized operation"),
+          @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response updatePagePermissions(@Context
+  HttpServletRequest request,
+                                        @Parameter(description = "Page reference", required = true)
+                                        @PathParam("pageRef")
+                                        String pageRef,
+                                        @Parameter(description = "Page new edit permission", required = true)
+                                        @QueryParam("editPermission")
+                                        String editPermission,
+                                        @Parameter(description = "Page new access permissions", required = true)
+                                        @QueryParam("accessPermissions")
+                                        String accessPermissions) {
+    try {
+      if (pageRef == null || editPermission == null || editPermission.isBlank() ||  accessPermissions == null || accessPermissions.isEmpty()) {
+        return Response.status(Response.Status.BAD_REQUEST).entity("params are mandatory").build();
+      }
+      PageContext page = layoutService.getPageContext(PageKey.parse(pageRef));
+      if(page == null){
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      if (!SiteNavigationUtils.canEditPage(page)) {
+        return Response.status(Response.Status.UNAUTHORIZED).build();
+      }
+      PageState pageState = page.getState();
+      List<String> accessPermissionsList = List.of(accessPermissions.split(","));
+      page.setState(new PageState(pageState.getDisplayName(), pageState.getDescription(), pageState.getShowMaxWindow(), pageState.getFactoryId(), accessPermissionsList, editPermission, pageState.getMoveAppsPermissions(),pageState.getMoveContainersPermissions()));
+      layoutService.save(page);
```

Comment:
we should check if we should use layoutService.save(PageContext pageContext, Page page) instead

Reply:
updation pageContext is sufficient